assets/js/index.js

```diff
@@ -0,0 +1,157 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+//function - get the password length from the user
+const getPasswordLength = () => {
+  //prompt user to give a userInput
+  //declare variable to receive user input - const -> not modified further down
+  const userInput = prompt(
+    "How long would you like your password to be? \n\nPlease enter a number between 8 and 128.\n",
+    ""
+  );
+  // check if userInput is a number
+  const passwordLength = parseInt(userInput);
+
+  // converts userInput into a number (or return NaN if not a number)
+  if (!isNaN(passwordLength) && passwordLength >= 8 && passwordLength <= 128) {
+    //if type and range are both true, the number is returned for storage in variable
+    return passwordLength;
+  }
+  //if type or range are not true, it sends an alert box with message
+  else {
+    alert(
+      "You must enter a number between 8 and 128. \n\nClick OK to start again."
+    );
+    //click ok on alert box to reload the page and start again
+    window.location.reload(true);
+    document.getElementById("password").reset();
+  }
+};
+
+//function - get the user preferences on which characters to include in password
+const getPasswordOptions = () => {
+  //step 1: getUserOption part
+  //declare variables for questions and strings of characters
+  const question = [
+    "Would you like to use lowercase letters in your password? \nClick 'OK' for yes and 'Cancel' for no.",
+    "Would you like to use uppercase letters in your password? \nClick 'OK' for yes and 'Cancel' for no.",
+    "Would you like to use numbers in your password? \nClick 'OK' for yes and 'Cancel' for no.",
+    "Would you like to use special characters in your password? \nClick 'OK' for yes and 'Cancel' for no.",
+  ];
+  const characters = [
+    "abcdefghijklmnopqrstuvwxyz",
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+    "0123456789",
+    " !\"#$%&'()*+,-./:;<=>?@[]^_`{|}~",
+  ];
+  //declare variable for result array of strings of characters - start with empty array - const -> only pushing in it, not modifying
+  const choiceArray = [];
+
+  //prompt the user with a question and get true/false response
+  //4 prompts - loop for 4 questions
+  for (let i = 0; i < question.length; i += 1) {
+    const getUserOption = confirm(question[i]); //const -> used only in this loop, not modified in the if statement
```

Comment:
I would have named this as `userOption`. The prefix of `get` suggests that something is a function which will return something.

```suggestion
    const userOption = confirm(question[i]); //const -> used only in this loop, not modified in the if statement
```